tests/test_selection/test_probe_feature_selection.py

```diff
@@ -0,0 +1,273 @@
+    )
+
+    assert sel.estimator == _estimator
+    assert sel.scoring == _scoring
+    assert sel.distribution == _distribution
+    assert sel.cv == _cv
+    assert sel.n_probes == _n_probes
+    assert sel.random_state == _random_state
+
+
+@pytest.mark.parametrize("_distribution", [3, "poisson", ["normal", "binary"], 2.22])
+def test_when_not_permitted_distribution(_distribution):
+    with pytest.raises(ValueError):
+        ProbeFeatureSelection(
+            estimator=DecisionTreeRegressor(),
+            distribution=_distribution,
+        )
+
+
+@pytest.mark.parametrize("_n_probes", [5, 7, 11])
+def test_when_not_permitted_n_probes_with_all_distribution(_n_probes):
+    with pytest.raises(ValueError):
+        ProbeFeatureSelection(
+            estimator=RandomForestClassifier(),
+            distribution="all",
+            n_probes=_n_probes,
+        )
+
+
+@pytest.mark.parametrize("_n_probes", ["tree", [False, 2], 101.1])
+def test_when_not_permitted_n_probes(_n_probes):
+    with pytest.raises(ValueError):
+        ProbeFeatureSelection(
+            estimator=DecisionTreeRegressor(),
+            n_probes=_n_probes,
+        )
+
+
+def test_fit_attributes(df_test):
+    X, y = df_test
+    y = column_or_1d(y, warn=True)
+
+    sel = ProbeFeatureSelection(
+        estimator=RandomForestClassifier(),
+        distribution="normal",
+        n_probes=2,
+        scoring="recall",
+        cv=3,
+        random_state=3,
+        confirm_variables=False,
+    )
+    sel.fit(X, y)
+
+    # expected results
+    expected_probe_features = {
+        "gaussian_probe_0": [5.366, 1.31, 0.289, -5.59, -0.832],
+        "gaussian_probe_1": [0.104, 3.396, -7.67, -0.807, -5.729],
+    }
+
+    expected_probe_features_df = pd.DataFrame(expected_probe_features)
+    expected_feature_importances = pd.Series(
+        data=[0.03, 0, 0, 0, 0.26, 0, 0.22, 0.33, 0.02, 0.12, 0, 0, 0, 0],
+        index=[
+            "var_0",
+            "var_1",
+            "var_2",
+            "var_3",
+            "var_4",
+            "var_5",
+            "var_6",
+            "var_7",
+            "var_8",
+            "var_9",
+            "var_10",
+            "var_11",
+            "gaussian_probe_0",
+            "gaussian_probe_1",
+        ],
+    )
+    assert sel.probe_features_.head().round(3).equals(expected_probe_features_df)
+    assert sel.features_to_drop_ == ["var_2", "var_10"]
+    assert sel.feature_importances_.round(2).equals(expected_feature_importances)
+
+
+def test_generate_probe_features(load_diabetes_dataset):
+    X, y = load_diabetes_dataset
+    X.columns = [f"var_{col}" for col in X.columns]
+
+    sel = ProbeFeatureSelection(
+        estimator=DecisionTreeRegressor(),
+        variables=None,
+        scoring="neg_mean_squared_error",
+        n_probes=6,
+        distribution="all",
+        cv=3,
+        random_state=1,
+        confirm_variables=False
+    )
+
+
+    sel.fit(X, y.values.ravel())
```

Comment:
This is doing what is expected and we can test the generation of the probe features, but there are a lot of other things happening as well. 

My idea was to call the method `_genearate_probe_features()` with the required parameters, it returns a df, capture the df in a variable and then assert if equal to the expected.